### PR TITLE
Fixes #23470 - Reset puppet server version to ensure smooth upgrades

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
@ekohl This might seem odd at first, but this is needed for example if you have Puppet 4 and move to the Puppet 5 repository while upgrading which I think is a supported path for us now that P5 is not drastic from P4.

File: hooks/pre_validations/31-upgrade-puppet.rb [hunk 1]
```diff
@@ -1,23 +1,7 @@
...
-  katello = Kafo::Helpers.module_enabled?(@kafo, 'katello')
-  foreman_proxy_content = Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_content')
-
-  fail_and_exit 'Puppet 4 to 5 upgrade is not currently supported for the chosen scenario.' unless katello || foreman_proxy_content
-
-  Kafo::Helpers.log_and_say :info, 'Resetting puppet params...'
+if app_value(:upgrade)
+  Kafo::Helpers.log_and_say :info, 'Resetting puppet server version param...'
 
   if !app_value(:noop)
```
Comment (ekohl) [inline hooks/pre_validations/31-upgrade-puppet.rb]: Maybe also make this `unless` rather than `if !`

File: hooks/boot/31-upgrade-puppet.rb [hunk 2]
```diff
@@ -1,6 +0,0 @@
-app_option(
-  '--upgrade-puppet',
-  :flag,
-  "Run the steps neccessary to upgrade from Puppet 3 to Puppet 4.",
```
Comment (ekohl) [inline hooks/boot/31-upgrade-puppet.rb]: Looks like we never updated this :)

**Comment by ekohl** [comment]
> @ekohl This might seem odd at first, but this is needed for example if you have Puppet 4 and move to the Puppet 5 repository while upgrading which I think is a supported path for us now that P5 is not drastic from P4.

As I mentioned: reinstalling the oauth gem is needed from puppet 4 to puppet 5.

**Comment by ehelms** [comment]
Updated to run this at all times and check the environment. Tested on an upgrade and worked.

**Pull request merged**